src/3d/materials/qgsphongtexturedmaterialsettings.cpp

```diff
@@ -106,56 +109,75 @@ Qt3DRender::QMaterial *QgsPhongTexturedMaterialSettings::toMaterial( QgsMaterial
       const QImage textureSourceImage = QgsApplication::imageCache()->pathAsImage( mDiffuseTexturePath, QSize(), true, 1.0, fitsInCache );
       ( void )fitsInCache;
 
-      if ( !textureSourceImage.isNull() )
+      // No texture image was provided.
+      // Fallback to QgsPhongMaterialSettings.
+      if ( textureSourceImage.isNull() )
       {
-        QgsImageTexture *textureImage = new QgsImageTexture( textureSourceImage );
-        Qt3DExtras::QDiffuseSpecularMaterial *material = new Qt3DExtras::QDiffuseSpecularMaterial;
-
-        Qt3DRender::QTexture2D *texture = new Qt3DRender::QTexture2D();
-        texture->addTextureImage( textureImage );
-
-        texture->wrapMode()->setX( Qt3DRender::QTextureWrapMode::Repeat );
-        texture->wrapMode()->setY( Qt3DRender::QTextureWrapMode::Repeat );
-        texture->wrapMode()->setZ( Qt3DRender::QTextureWrapMode::Repeat );
-
-        texture->setSamples( 4 );
-
-        texture->setGenerateMipMaps( true );
-        texture->setMagnificationFilter( Qt3DRender::QTexture2D::Linear );
-        texture->setMinificationFilter( Qt3DRender::QTexture2D::Linear );
-
-        material->setDiffuse( QVariant::fromValue( texture ) );
-
-        material->setSpecular( mSpecular );
-        material->setAmbient( mAmbient );
-        material->setShininess( mShininess );
-        material->setTextureScale( mTextureScale );
-
-        if ( context.isSelected() )
-        {
-          // update the material with selection colors
-          // TODO : dampen the color of diffuse texture
-          //      mat->setDiffuse( context.map().selectionColor() );
-          material->setAmbient( context.selectionColor().darker() );
-        }
-
-        return material;
-      }
-      else
-      {
-        Qt3DExtras::QPhongMaterial *material  = new Qt3DExtras::QPhongMaterial;
-        material->setAmbient( mAmbient );
-        material->setSpecular( mSpecular );
-        material->setShininess( mShininess );
-
-        if ( context.isSelected() )
-        {
-          // update the material with selection colors
-          material->setDiffuse( context.selectionColor() );
-          material->setAmbient( context.selectionColor().darker() );
-        }
+        QgsPhongMaterialSettings phongSettings = QgsPhongMaterialSettings();
+        phongSettings.setAmbient( mAmbient );
+        phongSettings.setDiffuse( QColor::fromRgbF( 0.7f, 0.7f, 0.7f, 1.0f ) ); // default diffuse color from QDiffuseSpecularMaterial
+        phongSettings.setOpacity( mOpacity );
+        phongSettings.setShininess( mShininess );
+        phongSettings.setSpecular( mSpecular );
+        Qt3DRender::QMaterial *material = phongSettings.toMaterial( technique, context );
         return material;
       }
+
+      Qt3DRender::QMaterial *material = new Qt3DRender::QMaterial;
```

Comment:
It may not be obvious to reader why we're not using QDiffuseSpecularMaterial here, worth adding a comment to explain - supposedly because it does not support opacity?